src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,95 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
```

Comment:
To leverage the full benefits of Java's type system and ensure type safety, it's recommended to use the generic type parameter T rather than a non-specific Object array for the internal storage of ArrayList. This approach enables the compiler to check for type compatibility at compile time, preventing potential runtime errors and making the code more maintainable